### PR TITLE
cli: add non-integrational tests for cli packages

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Added a set of naive tests for different cli packages.

### CODE REVIEW

**Comment by fyrchik** [comment]
I think we also need some tests which will check correctness of our flag methods from the viewpoint of cli library we use (which is the main thing we are interested in). All other tests are more like internal details of the implementation.
Something like this:
```
func TestAddress(t *testing.T) {
	f := flag.NewFlagSet("", flag.ContinueOnError)
	f.SetOutput(ioutil.Discard) // don't pollute test output
	addr := AddressFlag{Name: "addr, a"}
	addr.Apply(f)
	require.NoError(t, f.Parse([]string{"--addr", "NRHkiY2hLy5ypD32CKZtL6pNwhbFMqDEhR"}))
	require.Equal(t, "NRHkiY2hLy5ypD32CKZtL6pNwhbFMqDEhR", f.Lookup("a").Value.String())
	require.Error(t, f.Parse([]string{"--addr", "kek"}))
}
```

File: pkg/core/storage/store.go [hunk 1]
```diff
@@ -91,6 +92,8 @@ func NewStore(cfg DBConfiguration) (Store, error) {
 		store, err = NewBoltDBStore(cfg.BoltDBOptions)
 	case "badgerdb":
 		store, err = NewBadgerDBStore(cfg.BadgerDBOptions)
+	default:
+		return nil, fmt.Errorf("uncknown storage: %s", cfg.Type)
```
Comment (roman-khimov) [inline pkg/core/storage/store.go:96]: unknown

File: cli/server/server_test.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+}
+
+func TestLogo(t *testing.T) {
+	require.Equal(t, `
+    _   ____________        __________
+   / | / / ____/ __ \      / ____/ __ \
+  /  |/ / __/ / / / /_____/ / __/ / / /
+ / /|  / /___/ /_/ /_____/ /_/ / /_/ /
+/_/ |_/_____/\____/      \____/\____/
+`, logo())
```
Comment (roman-khimov) [inline cli/server/server_test.go:227]: My favorite!

File: cli/server/server.go [hunk 3]
```diff
@@ -147,7 +147,7 @@ func initBCWithMetrics(cfg config.Config, log *zap.Logger) (*core.Blockchain, *m
 	if err != nil {
 		return nil, nil, nil, cli.NewExitError(err, 1)
 	}
-	configureAddresses(cfg.ApplicationConfiguration)
+	configureAddresses(&cfg.ApplicationConfiguration)
```
Comment (roman-khimov) [inline cli/server/server.go:150]: OMG! That's a completely separate fix.

File: cli/smartcontract/smart_contract_test.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/stretchr/testify/require"
+	"github.com/urfave/cli"
+)
+
+func TestInitSmartContract(t *testing.T) {
+	contractPath := "testContract"
```
Comment (roman-khimov) [inline cli/smartcontract/smart_contract_test.go:17]: `ioutil.TempDir`?

File: cli/server/server_test.go [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+	})
+}
+
+func TestRestoreDB(t *testing.T) {
+	testDump := "./test_dump.acc"
+	saveDump := "./dump.acc"
+	defer func() {
+		os.RemoveAll("./chains/")
+		os.RemoveAll(testDump)
+		os.RemoveAll(saveDump)
```
Comment (roman-khimov) [inline cli/server/server_test.go:126]: `ioutil.TempDir`? Maybe change working directory to that temp dir. But then wipe it and forget it.

**Review by roman-khimov — COMMENTED** [review]
TempDir/TempFile, try it to use them in every case you need some throw-away entity.

**Comment by AnnaShaleva** [comment]
> I think we also need some tests which will check correctness of our flag methods from the viewpoint of cli library we use (which is the main thing we are interested in). All other tests are more like internal details of the implementation.
> Something like this:
> 
> ```
> func TestAddress(t *testing.T) {
> 	f := flag.NewFlagSet("", flag.ContinueOnError)
> 	f.SetOutput(ioutil.Discard) // don't pollute test output
> 	addr := AddressFlag{Name: "addr, a"}
> 	addr.Apply(f)
> 	require.NoError(t, f.Parse([]string{"--addr", "NRHkiY2hLy5ypD32CKZtL6pNwhbFMqDEhR"}))
> 	require.Equal(t, "NRHkiY2hLy5ypD32CKZtL6pNwhbFMqDEhR", f.Lookup("a").Value.String())
> 	require.Error(t, f.Parse([]string{"--addr", "kek"}))
> }
> ```

Added a couple of such tests for our custom address and fixed8 flags.

File: cli/server/server_test.go (hunk 2 above)
Comment (AnnaShaleva) [reply cli/server/server_test.go]: Removed it(

File: cli/server/server_test.go [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+	})
+}
+
+func TestRestoreDB(t *testing.T) {
+	testDump, err := ioutil.TempFile("./", "*.acc")
+	require.NoError(t, err)
+	saveDump, err := ioutil.TempFile("./", "*.acc")
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, os.RemoveAll("./chains/"))
```
Comment (roman-khimov) [inline cli/server/server_test.go:128]: Maybe
```
d := ioutil.TempDir()
testDump := "file1.acc"
saveDump := "file2.acc"
os.Chdir(d)
defer func() {
    os.Chdir("..")
    os.RemoveAll(d)
}
```
?

File: cli/smartcontract/smart_contract_test.go (hunk 4 above)
Comment (AnnaShaleva) [reply cli/smartcontract/smart_contract_test.go]: Now fixed.

File: cli/server/server_test.go (hunk 6 above)
Comment (AnnaShaleva) [reply cli/server/server_test.go]: That's better, fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**